src/Domain/Entity/TermsVersion.php

```diff
@@ -0,0 +1,219 @@
+     */
+    protected $terms;
+
+    /**
+     * @var ArrayCollection<int, TermsVersionSignature>
+     *
+     * @ORM\OneToMany(targetEntity="RichId\TermsModuleBundle\Domain\Entity\TermsVersionSignature", mappedBy="version")
+     */
+    protected $signatures;
+
+    public function __construct()
+    {
+        $this->signatures = new ArrayCollection();
+    }
+
+    public function getId(): ?int
+    {
+        return $this->id;
+    }
+
+    public function getVersion(): ?int
+    {
+        return $this->version;
+    }
+
+    public function setVersion(int $version): self
+    {
+        $this->version = $version;
+
+        return $this;
+    }
+
+    public function isEnabled(): bool
+    {
+        return $this->isEnabled;
+    }
+
+    public function enable(): self
+    {
+        $this->isEnabled = true;
+
+        return $this;
+    }
+
+    public function getTitle(): ?string
+    {
+        return $this->title;
+    }
+
+    public function setTitle(string $title): self
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    public function getContent(): ?string
+    {
+        return $this->content;
+    }
+
+    public function setContent(string $content): self
+    {
+        $this->content = $content;
+
+        return $this;
+    }
+
+    public function getPublicationDate(): ?\DateTime
+    {
+        return $this->publicationDate;
+    }
+
+    public function setPublicationDate(?\DateTime $publicationDate): self
+    {
+        $this->publicationDate = $publicationDate;
+
+        return $this;
+    }
+
+    public function getTerms(): Terms
+    {
+        return $this->terms;
+    }
+
+    public function setTerms(Terms $terms): self
+    {
+        $this->terms = $terms;
+
+        return $this;
+    }
+
+    /** @return ArrayCollection<int, TermsVersionSignature> */
+    public function getSignatures(): Collection
+    {
+        return $this->signatures;
+    }
+
+    public function addSignature(TermsVersionSignature $signature): self
+    {
+        $this->signatures->add($signature);
+
+        return $this;
+    }
+
+    public function removeSignature(TermsVersionSignature $signature): self
+    {
+        $this->signatures->removeElement($signature);
+
+        return $this;
+    }
+
+    public static function buildDefaultVersion(Terms $terms): self
+    {
+        $entity = new self();
+
+        $entity->setTerms($terms);
+        $entity->setVersion(1);
+        $terms->addVersion($entity);
+
+        return $entity;
+    }
+
+    public static function buildFromCopy(TermsVersion $termsVersion): self
+    {
+        $terms = $termsVersion->getTerms();
+        $lastVersion = $terms->getLatestVersion();
+        $nextVersion = $lastVersion !== null ? $lastVersion->getVersion() + 1 : 1;
+
+        $entity = new self();
+
+        $entity->setTerms($terms);
+        $entity->setVersion($nextVersion);
+        $entity->setTitle($termsVersion->getTitle()); /* @phpstan-ignore-line */
```

Comment:
Add a comment to explain why this line is excluded from static analysis (and same remark for all the others).

Why is it excluded from static analysis? Is there a specific rule we want to ignore here (and can we ignore only that rule?) ?